source/javascripts/components/_Variable.js.erb

```diff
@@ -83,6 +90,18 @@ angular.module("BitriseWorkflowEditor").factory("Variable", function(Meta) {
 		return true;
 	};
 
+	Variable.isValidValue = function(value, shouldBeNotEmpty) {
+		if (value === undefined) {
+			return undefined;
+		}
+
+		if (!shouldBeNotEmpty || value === null || !value || value.replace(/\s/g, '').length < 1) {
```

Comment:
Isn't `null` the value when it is just not fetched with the eye icon? It should be valid in that case. Also, do a better regexp: beginning of string + any number of whitespaces (even 0) + end of string. And in this case, you can leave the `!value! part, because the regexp will cover that too.